main.cpp

```diff
@@ -936,12 +943,27 @@ int main(int argc, char ** argv) {
         printf(ANSI_COLOR_YELLOW);
     }
 
+    if (params.embedding){
+        printf("got right before second call.\n");
+        const int64_t t_start_us = ggml_time_us(); //HERE
+        if (!llama_eval(model, params.n_threads, n_past, embd, logits, mem_per_token, true)) {
```

Comment:
You're not calling `llama_eval` for remaining tokens and doing other checks.

Reply:
the sentence embedding depends on the input, we shouldn't have to generate extra tokens later. Though I may be wrong. But changing that wouldn't solve the failed execution error.

Reply:
Are you trying to capture input text's embeddings or predicted text embeddings?

Reply:
Given the PR it is doing the input text's embeddings

Reply:
The vector of tokens that you are sending for evaluation is empty. Please ensure that it is set correctly right before evaluating it, and also add a check to verify that it is not empty.

```c++
embd = embd_inp;

if (embd.size() > 0) {
   if (!llama_eval(model, params.n_threads, n_past, embd, logits, mem_per_token, true)) {
      fprintf(stderr, "Failed to predict\n");
      return 1;
   }
}
```

Reply:
Let me know if the above fixes the runtime crash

Reply:
That solved it! My version is working but I see merge conflicts, I am testing the merge now

Reply:
Merge successful. Now we have embeddings! This should be ready to merge into main.